Skyra.Core/Cache/Models/Prompts/CoreRichDisplay.cs

```diff
@@ -0,0 +1,315 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+using Skyra.Core.Utils;
+using Spectacles.NET.Types;
+
+namespace Skyra.Core.Cache.Models.Prompts
+{
+	public sealed class CoreRichDisplay : ICorePromptState, ICorePromptStateReaction
+	{
+		public CoreRichDisplay(ulong authorId, ulong messageId)
+		{
+			AuthorId = authorId;
+			MessageId = messageId;
+			InternalTemplate = new CoreMessageEmbed();
+			InformationPage = null;
+			AllowedEmojis = new (CoreRichDisplayReactionType, string)[0];
+			Emojis = new CoreRichDisplayEmojis();
+			FooterEnabled = false;
+			FooterPrefix = null;
+			FooterSuffix = null;
+			Context = new CoreMessageEmbed[0];
+			PagePosition = 0;
+		}
+
+		[JsonConstructor]
+		public CoreRichDisplay(ulong authorId, ulong messageId, [CanBeNull] CoreMessageEmbed[] context,
+			CoreMessageEmbed?
+				informationPage, int pagePosition, (CoreRichDisplayReactionType, string)[] allowedEmojis)
+		{
+			AuthorId = authorId;
+			MessageId = messageId;
+			AllowedEmojis = allowedEmojis;
+			InternalTemplate = new CoreMessageEmbed();
+			InformationPage = informationPage;
+			Emojis = new CoreRichDisplayEmojis();
+			FooterEnabled = false;
+			FooterPrefix = null;
+			FooterSuffix = null;
+			Context = context ?? new CoreMessageEmbed[0];
```

Comment:
fairly sure the nullability operator here will always return false, because you're assigning the array in the constructor.

Reply:
As a matter of a fact you're right, I'll fix that once I'm on PC.